snapcraft/internal/parser.py

```diff
@@ -240,9 +241,9 @@ def _process_index(output):
 
     # split the wiki into an array of entries to allow the parser to
     # proceed when invalid yaml is found.
-    entries = output.split(b'\n---\n')
+    entries = re.split(r'\r*\n---\r*\n', output.decode(), flags=re.MULTILINE)
```

Comment:
I would read line by line in the decoded already and looks for an empty newline it might be less error prone.